executors/keptn/pkg/keptn/keptn.go

```diff
@@ -0,0 +1,329 @@
+	}
+
+	return nil
+}
+func (k *Keptn) CreateService(service, project string) error {
+	if _, kErr := k.apiHandler.CreateService(project, apimodels.CreateService{
+		ServiceName: &service,
+	}); kErr != nil {
+		return fmt.Errorf("failed to create service with err: %v", kErr.GetMessage())
+	}
+
+	return nil
+}
+
+func (k *Keptn) AddResourceToAllStages(service, project, resourceName, resourceContent string) error {
+	stages, err := k.getProjectStages(project)
+	if err != nil {
+		return err
+	}
+
+	for _, stage := range stages {
+		if err := k.AddResourceToStage(service, project, stage.StageName, resourceNameToURI(resourceName), resourceContent); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (k *Keptn) AddResourceToStage(service, project, stage, resourceURI, resourceContent string) error {
+	// encodedResourceContent := base64.StdEncoding.EncodeToString([]byte(resourceContent))
+	encodedResourceContent := resourceContent
+	resource := &apimodels.Resource{
+		ResourceContent: encodedResourceContent,
+		ResourceURI:     &resourceURI,
+	}
+
+	if _, err := k.resourceHandler.CreateServiceResources(project, stage, service, []*apimodels.Resource{resource}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (k *Keptn) ConfigureMonitoring(project, service, monitoringType string) error {
+	configureMonitoringEventData := &keptnlib.ConfigureMonitoringEventData{
+		Type:    monitoringType,
+		Project: project,
+		Service: service,
+	}
+
+	source, _ := url.Parse("https://github.com/keptn/keptn/cli#configuremonitoring")
+
+	sdkEvent := cloudevents.NewEvent()
+	sdkEvent.SetID(uuid.New().String())
+	sdkEvent.SetType(keptnlib.ConfigureMonitoringEventType)
+	sdkEvent.SetSource(source.String())
+	sdkEvent.SetDataContentType(cloudevents.ApplicationJSON)
+	sdkEvent.SetData(cloudevents.ApplicationJSON, configureMonitoringEventData)
+
+	eventByte, err := json.Marshal(sdkEvent)
+	if err != nil {
+		return fmt.Errorf("failed to marshal cloud event. %s", err.Error())
+	}
+
+	apiEvent := apimodels.KeptnContextExtendedCE{}
+	err = json.Unmarshal(eventByte, &apiEvent)
+	if err != nil {
+		return fmt.Errorf("failed to map cloud event to API event model. %s", err.Error())
+	}
+
+	_, kErr := k.apiHandler.SendEvent(apiEvent)
+	if err != nil {
+		return fmt.Errorf("sending configure-monitoring event was unsuccessful. %s", *kErr.Message)
+	}
+
+	return nil
+}
+
+func (k *Keptn) GetEvents(service, project, keptnCtx string) error {
+	filter := &apiutils.EventFilter{
+		Project:      project,
+		Service:      service,
+		KeptnContext: keptnCtx,
+		EventType:    "sh.keptn.event.evaluation.finished",
+	}
+
+	eventsCtx, kErr := k.eventHandler.GetEvents(filter)
+	if kErr != nil {
+		return fmt.Errorf("failed to get events for keptn context %s. %s", keptnCtx, *kErr.Message)
+	}
+
+	if len(eventsCtx) != 1 {
+		return fmt.Errorf("expected to see one event of type %s", filter.EventType)
+	}
+
+	if dataMap, ok := eventsCtx[0].Data.(map[string]interface{}); ok {
```

Comment:
Does keptn have a struct format of this so we don't have to use `interface{}` and type-casting?

Reply:
unfortunately, the data is an opaque blob of data 😞 .